variables.tf

```diff
@@ -0,0 +1,109 @@
+# Variables for ALL modules
+variable "metal_auth_token" {
+  description = "Equinix Metal user api token."
+  type        = string
+  sensitive   = true
+}
+
+variable "metal_project_id" {
+  description = "Project ID"
+  type        = string
+}
+
+variable "metal_organization_id" {
+  type        = string
+  description = "Equinix Metal organization id"
+}
+
+# Variables for "invite-from-csv" module
+variable "enable_project_setup" {
+  type        = bool
+  description = "Enable Project Setup module"
+  default     = false
+}
+
+variable "csv_file" {
+  type        = string
+  description = "Path to a CSV file containing a list of projects to provision: email,metro,plan. Email address is used as the project name and the collaborator. Metro and plan are used to provision the project."
+  default     = "users.csv"
+}
+
+variable "send_invites" {
+  type        = bool
+  description = "Wether Collaborator invitations should be sent. This could be toggled after a successful provision to prevent sending invitations to a project that could be deleted during a reprovision"
+  default     = true
+}
+
+# Variables for "metal" module
+variable "enable_metal" {
+  type        = bool
+  description = "Enable Metal module"
+  default     = false
+}
+
+variable "metal_device_count" {
+  description = "Number of devices to provision"
+  type        = number
+  default     = 1
+}
+
+variable "metal_metro" {
+  description = "Number of devices to provision"
+  type        = string
+  default     = "sv"
+}
+
+variable "metal_os" {
+  description = "OS to provision"
+  type        = string
+  default     = "ubuntu_20_04"
+}
+
+variable "metal_billing_cycle" {
+  description = "Billing cycle"
+  type        = string
+  default     = "hourly"
+}
+
+variable "metal_device_type" {
+  description = "Equinix Metal device type to metal nodes"
+  type        = string
+  default     = "m3.small.x86"
+}
+
+# Variables for "eksa" module
+variable "enable_eksa" {
+  type        = bool
+  description = "Enable EKS-A module"
+  default     = false
+}
+
+variable "eksa_cluster_name" {
```

Comment:
These variables should follow the addons pattern and use something like an object called ekas_config (I didn't check the naming convention).

Reply:
Same comment about metal_* variables in this file.